processes/process_utils.c

```diff
@@ -87,21 +87,29 @@ pid_t reapChildren() {
 	}
 }
 
+// Here memory is allocated for array of strings
+// Memory for string elements in arrat is not allocated here
+// Look for `C.CString(s)` in func convertToCStrings for allocation
 char **makeCStringArray(int n) {
-	char **ret = (char **) malloc(sizeof(char *) * n);
-	memset(ret,0,sizeof(char *) * n);
+	int size = sizeof(char *) * n;
+	char **ret = (char **) malloc(size);
+	if(ret) {
+		memset(ret, 0, size);
+	}
 	return ret;
 }
 
+// Free all the memory allocated in array elements before freeing the array itself
 void freeCStringArray(char **a) {
 	int z = 0;
 	if(a) {
 		char *s = a[z];
 		while(s) {
 			free(s);
-			z++; s = a[z];
+			z++;
+            s = a[z];
```

Comment:
Hmm, is this weird spacing due to '\t' vs ' ' (tabs vs spaces)?

Reply:
yes.. tabs vs spaces and editor configs - PR #13 is for formatting fixes

Reply:
go fmt will uses tabs. Another irritation of the Golang tribe.
https://stackoverflow.com/questions/19094704/indentation-in-go-tabs-or-spaces
https://thenewstack.io/spaces-vs-tabs-a-20-year-debate-and-now-this-what-the-hell-is-wrong-with-go/

Reply:
recommend you all use `go fmt` with your editor. VSCode has a good plugin for Go which will do this.